### PR TITLE
Feature/implement crud actions for rewards

### PR DESCRIPTION
**Pull request by Webstero**
All features work and I improved views by the use bootstrap.

### CODE REVIEW

File: app/models/reward.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Reward < ApplicationRecord
+  validates :title, :description, presence: true, allow_nil: false
```
Comment (ciejj) [inline app/models/reward.rb]: with `presence: true` nil and an empty string fails validation, so no need to add `allow_nil: false` 
```suggestion
  validates :title, :description, presence: true
```

File: app/models/reward.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+  validates :price, numericality: { greater_than_or_equal_to: 1 }, presence: true, allow_nil: false
```
Comment (ciejj) [inline app/models/reward.rb:4]:
```suggestion
  validates :price, numericality: { greater_than_or_equal_to: 1 }, presence: true
```

File: app/views/admins/rewards/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+</p>
+
+<p>
+  <strong>Creation time:</strong>
+  <%= @reward.created_at%>
+</p>
+
+
+
+
```
Comment (ciejj) [inline app/views/admins/rewards/show.html.erb]: unnecessary empty lines

File: db/seeds.rb [hunk 4]
```diff
@@ -8,12 +8,19 @@
...
 1.upto(2) do |i|
   Admin.create!(email: "admin#{i}@test.com", password: 'password')
 end
+
+
+1.upto(5) do |i|
+  reward = Reward.create! title: "Reward #{i}",
+    description: Faker::Superhero.power,
+    price: Faker::Number.decimal(l_digits: 3, r_digits: 2)
+end
```
Comment (ciejj) [inline db/seeds.rb]: missing newline at the end of the file

File: db/seeds.rb [hunk 5]
```diff
@@ -8,12 +8,19 @@
...
+    giver: Employee.where("id = ? OR id = ?", Employee.first.id, Employee.second.id).sample, 
+    receiver: Employee.where.not("id = ? OR id = ?", Employee.first.id, Employee.second.id).sample, 
+    company_value_id: CompanyValue.all.sample[:id]
 end 
 
 1.upto(2) do |i|
   Admin.create!(email: "admin#{i}@test.com", password: 'password')
 end
+
+
```
Comment (ciejj) [inline db/seeds.rb]: unnecessary empty line

File: db/seeds.rb [hunk 6]
```diff
@@ -8,12 +8,19 @@
 
 1.upto(9) do
   kudo = Kudo.create! title: Faker::Name.unique.name, 
-  content: Faker::Hacker.say_something_smart, 
-  giver: Employee.where("id = ? OR id = ?", Employee.first.id, Employee.second.id).sample, 
-  receiver: Employee.where.not("id = ? OR id = ?", Employee.first.id, Employee.second.id).sample, 
-  company_value_id: CompanyValue.all.sample[:id]
+    content: Faker::Hacker.say_something_smart,
```
Comment (ciejj) [inline db/seeds.rb:11]: its not an error, but I would wrap those attributes in the brackets for the sake of consistency

File: spec/features/admin_process/admin_reward_process/admin_create_reward_spec.rb [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    click_link 'New Reward'
+    fill_in 'Title', with: reward.title
+    fill_in 'Description', with: reward.description
+    fill_in 'Price', with: reward.price
+    click_button 'Create Reward'
+    expect(page).to have_content 'Reward was successfully created.'
+    expect(page).to have_current_path(admins_rewards_path)
+  end
+
+  it 'Create Reward without a title' do
```
Comment (ciejj) [inline spec/features/admin_process/admin_reward_process/admin_create_reward_spec.rb:24]: not an error, but you have already covered all validations in the model spec - there is no need to test it again in the system spec (which takes longer)

File: spec/features/admin_process/admin_reward_process/admin_create_reward_spec.rb [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+RSpec.describe 'Admin create reward', type: :feature do
+  let(:admin) { create(:admin) }
+  let(:reward) { create(:reward) }
+
+  before do
+    visit root_path
+  end
+
+  it 'Create Reward with correct credentials' do
+    click_link 'Sign in as Admin'
```
Comment (ciejj) [inline spec/features/admin_process/admin_reward_process/admin_create_reward_spec.rb]: consider moving login action to `before` action

File: spec/features/admin_process/admin_reward_process/admin_listning_rewards_spec.rb [hunk 9]
```diff
@@ -0,0 +1,26 @@
+require 'rails_helper'
+require 'capybara/rspec'
+
+RSpec.describe 'Admin listing reward', type: :feature do
+  let(:admin) { create(:admin) }
+  let!(:reward) { create(:reward) }
```
Comment (ciejj) [inline spec/features/admin_process/admin_reward_process/admin_listning_rewards_spec.rb:6]: I would create 2 rewards, and check that it displays both of them :)

**Review by ciejj — APPROVED** [review]
Nice one 👍

File: spec/features/admin_process/admin_reward_process/admin_create_reward_spec.rb (hunk 7 above)
Comment (Webstero) [reply spec/features/admin_process/admin_reward_process/admin_create_reward_spec.rb:24]: I just wanted to check content about validations.

**Pull request merged**